terraform/nextstrain_batch.tf

```diff
@@ -0,0 +1,130 @@
+variable "nextstrain-batch-params" {
+  type  = map
+  default = {
+    timeout = 14400
+  }
+}
+
+resource "aws_batch_job_definition" "nextstrain-job-definition" {
+  name = "nextstrain-job-definition"
+  type = "container"
+  parameters = var.nextstrain-batch-params
+
+  container_properties = <<CONTAINER_PROPERTIES
+{
+    "command": [],
+    "image": "nextstrain/base:latest",
+    "memory": 1024,
+    "vcpus": 1,
+    "jobRoleArn": "${aws_iam_role.nextstrain-jobs-role.arn}"
+}
+CONTAINER_PROPERTIES
+}
+
+
+data "aws_availability_zones" "available" {}
+
+resource "aws_default_vpc" "default" {}
+
+data "aws_internet_gateway" "default" {
+  filter {
+    name   = "attachment.vpc-id"
+    values = [aws_default_vpc.default.id]
+  }
+}
+
+
+resource "aws_default_route_table" "default" {
+  default_route_table_id = aws_default_vpc.default.default_route_table_id
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = data.aws_internet_gateway.default.id
+  }
+}
+
+
+resource "aws_default_subnet" "default" {
+  count = length(split(",", join(",", flatten(data.aws_availability_zones.available.*.names))))
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+}
+
+
+resource "aws_security_group" "nextstrain-batch-security-group" {
+  name = "nextstrain_aws_batch_compute_environment_security_group"
+  description = "security group for nextstrain batch"
+  vpc_id = aws_default_vpc.default.id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+
+# for increased disk space
+resource "aws_launch_template" "nextstrain-launch-template" {
+  name = "nextstrain-launch-template"
+
+  block_device_mappings {
+    device_name = "/dev/xvdcz"
+
+    ebs {
+      volume_size = 200
+      volume_type = "gp2"
+      delete_on_termination = true
+    }
+  }
+  user_data = base64encode(file("${path.module}/advanced_options/user_data.txt"))
```

Comment:
ah, this is my bad, i misread and didn't realize the field was literally `user_data`.  bad choice on aws's part!

Reply:
haha yeah